src/core/mi_memory_resource.cc

```diff
@@ -40,4 +47,37 @@ void MiMemoryResource::do_deallocate(void* ptr, std::size_t size, std::size_t al
   mi_free_size_aligned(ptr, size, align);
 }
 
+void* FiberStackResource::do_allocate(size_t size, size_t align) {
+  // deliberately ignore alignment because mimalloc overprovisions memory by align -1 bytes.
+  void* res = mi_malloc(size);
+  if (res == nullptr) {
+    throw bad_alloc();
+  }
+
+  // in theory res does not have to be page aligned. In that case madvise will silently fail.
+  DCHECK(is_page_aligned(res)) << res;
+
+  madvise((void*)res, 4096, MADV_DONTNEED);
```

Comment:
I suggest to provide ability to choose between normal and huge pages

Reply:
see my previous point, we do not control this, it's hidden by mi_malloc.